dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,569 @@
+
+        this.doRememberMe(req, res, user, jwtMaxAge);
+    } // doRememberMe.
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) {
+
+        try {
+
+            this.processJsonWebToken(req, res, user, maxAge);
+        } catch (Exception e) {
+
+            Logger.debug(this, e.getMessage(), e);
+        }
+    } // doRememberMe.
+
+
+    private boolean _doCookieLogin(String encryptedId, HttpServletRequest request,
+                    HttpServletResponse response) {
+
+        try {
+            String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
+            /* Custom Code */
+            User user = null;
+            if (Validator.isEmailAddress(decryptedId))
+                user = userAPI.loadByUserByEmail(decryptedId, systemUser, false);
+            else
+                user = userAPI.loadUserById(decryptedId, systemUser, false);
+            /* End of Custom Code */
+            try {
+                String userName = (authByEmail) ? user.getEmailAddress() : user.getUserId();
+
+                return _doLogin(userName, null, true, request, response, true);
+            } catch (Exception e) { // $codepro.audit.disable logExceptions
+                SecurityLogger.logInfo(this.getClass(),
+                                "An invalid attempt to login (No user found) from IP: "
+                                                + request.getRemoteAddr() + " :  " + e);
+
+                return false;
+            }
+        } catch (Exception e) {
+            SecurityLogger.logInfo(this.getClass(), "Auto login failed (No user found) from IP: "
+                            + request.getRemoteAddr() + " :  " + e);
+
+            if (useCASLoginFilter) {
+                String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
+                Logger.info(this.getClass(),
+                                "Try to retrieve user from LDAP/CAS with id: " + decryptedId);
+                User newUser = CASAuthUtils.loadUserFromLDAP(decryptedId);
+
+                if (UtilMethods.isSet(newUser)) {
+                    User user = null;
+
+                    try {
+                        user = (authByEmail)
+                                        ? userAPI.loadByUserByEmail(decryptedId, systemUser, false)
+                                        : userAPI.loadUserById(decryptedId, systemUser, false);
+
+
+                        String userIdFromCAS = (String) request.getSession(false)
+                                        .getAttribute("edu.yale.its.tp.cas.client.filter.user");
+
+                        if (UtilMethods.isSet(userIdFromCAS)) {
+                            CASAuthUtils.setUserValuesOnSession(user, request, response, true);
+                        }
+
+                        return true;
+
+                    } catch (Exception ex) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)